### PR TITLE
feat(j-s): fetch parent case files

### PR DESCRIPTION
**Pull request by Arnarkari93**
[Asana - Framlenging - gögn send dómnum í fyrra máli](https://app.asana.com/0/1199153462262248/1202037182089588/f)
### Island-ui
-  make onRemoveClick optional

### Judicial-system
- show files of the parent case
- add tests for displaying files of the parent case
- move CaseFilesForm.tsx to CaseFiles.tsx
- move StepFiveForm.tsx to StepFive.tsx

## Why
- Better overview of case files related to current case

## Screenshots / Gifs

Screenshot 2022-06-14 at 15 15 19

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Prosecutor/RestrictionCase/StepFive/StepFive.tsx [hunk 1]
```diff
@@ -90,12 +235,251 @@ export const StepFive: React.FC = () => {
...
+          <CaseInfo
+            workingCase={workingCase}
+            userRole={user?.role}
+            showAdditionalInfo
+          />
+        </Box>
+        <ParentCaseFiles
+          caseType={workingCase.type}
+          files={workingCase.parentCase?.caseFiles}
+        />
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/routes/Prosecutor/RestrictionCase/StepFive/StepFive.tsx:254]: This is what I am adding

**Review by gudjong — CHANGES_REQUESTED** [review]
This looks good but `caseFiles` is missing from the `gql` query used by `FormProvider`.

File: apps/judicial-system/web-e2e/src/integration/Prosecutor/InvestigationCases/caseFiles.spec.ts [hunk 2]
```diff
@@ -20,4 +25,20 @@ describe(`${IC_CASE_FILES_ROUTE}/:id`, () => {
 
     cy.get('[name=fileUpload]').attachFile('lorem-ipsum.txt')
   })
+
+  it.only('should list of files for parent case', () => {
```
Comment (oddsson) [inline apps/judicial-system/web-e2e/src/integration/Prosecutor/InvestigationCases/caseFiles.spec.ts:29]:
```suggestion
  it('should list of files for parent case', () => {
```

File: apps/judicial-system/web-e2e/src/integration/Prosecutor/RestrictionCases/caseFiles.spec.ts [hunk 3]
```diff
@@ -26,4 +27,20 @@ describe(`${STEP_FIVE_ROUTE}/:id`, () => {
 
     cy.get('[name=fileUpload]').attachFile('lorem-ipsum.txt')
   })
+
+  it.only('should list of files for parent case', () => {
```
Comment (oddsson) [inline apps/judicial-system/web-e2e/src/integration/Prosecutor/RestrictionCases/caseFiles.spec.ts:31]:
```suggestion
  it('should list of files for parent case', () => {
```

File: apps/judicial-system/web/messages/InvestigationCases/Prosecutor/caseFiles.ts [hunk 4]
```diff
@@ -8,6 +8,15 @@ export const icCaseFiles = {
       'Notaður sem titill á rannsóknargagna skrefi í rannsóknarheimildum.',
   }),
   sections: {
+    parentCaseFiles: defineMessages({
+      heading: {
+        id:
+          'judicial.system.investigation_cases:case_files.parentCaseFiles.heading',
```
Comment (oddsson) [inline apps/judicial-system/web/messages/InvestigationCases/Prosecutor/caseFiles.ts:14]: We usually write the ids in_this_format. Please consider changing parentCaseFiles to parent_case_files

File: apps/judicial-system/web/messages/RestrictionCases/Prosecutor/caseFiles.ts [hunk 5]
```diff
@@ -8,6 +8,15 @@ export const rcCaseFiles = {
       'Notaður sem titill á rannsóknargagna skrefi í gæsluvarðhalds- og farbannsmálum.',
   }),
   sections: {
+    parentCaseFiles: defineMessages({
+      heading: {
+        id:
+          'judicial.system.restriction_cases:case_files.parentCaseFiles.heading',
```
Comment (oddsson) [inline apps/judicial-system/web/messages/RestrictionCases/Prosecutor/caseFiles.ts:14]: Same ☝️

**Review by oddsson — CHANGES_REQUESTED** [review]
Couple of things...

**Review by oddsson — APPROVED** [review]

**Review by gudjong — APPROVED** [review]

**Pull request merged**